core/org.eclipse.cdt.core/parser/org/eclipse/cdt/internal/core/dom/parser/cpp/semantics/EvalConstructor.java

```diff
@@ -160,11 +160,15 @@ public ICPPEvaluation computeForFunctionCall(ActivationRecord callSiteRecord, Co
 			return this;
 		}
 		final ICPPClassType classType = (ICPPClassType) unwrappedType;
-		final CompositeValue compositeValue = CompositeValue.create(classType);
+		final IValue classObject = CompositeValue.create(classType);
 		ICPPEvaluation[] argList = evaluateArguments(fArguments, callSiteRecord, context);
-		EvalFixed constructedObject = new EvalFixed(fType, ValueCategory.PRVALUE, compositeValue);
+		EvalFixed constructedObject = new EvalFixed(fType, ValueCategory.PRVALUE, classObject);
 		CPPVariable binding = new CPPVariable(TEMP_NAME);
 
+		if (!(classObject instanceof CompositeValue compositeValue)) {
```

Comment:
This variable is unused

```suggestion
		if (!(classObject instanceof CompositeValue)) {
```

Reply:
Sorry, I didn't know that the code could do what it did. That variable was indeed used, fix coming in a moment.